python/ray/_private/services.py

```diff
@@ -118,8 +118,21 @@ def address(ip_address, port):
     return ip_address + ":" + str(port)
 
 
-def new_port():
-    return random.randint(10000, 65535)
+def new_port(lower_bound=10000, upper_bound=65535, blacklist=None):
+    if not blacklist:
+        blacklist = set()
+    port = random.randint(lower_bound, upper_bound)
+    retry = 0
+    while port in blacklist:
+        if retry > 20:
+            break
+        random.randint(lower_bound, upper_bound)
```

Comment:
I think the `port` reassignment on retry is missing here.

```suggestion
        port = random.randint(lower_bound, upper_bound)
```